Firestore/core/src/core/query.cc

```diff
@@ -52,6 +52,15 @@ Query::Query(ResourcePath path, std::string collection_group)
           std::make_shared<const std::string>(std::move(collection_group))) {
 }
 
+Query Query::Clone() const {
+  Query copy(*this);
+  copy.filters_ = immutable::AppendOnlyList<Filter>();
+  for (const Filter& filter : filters_) {
+    copy.filters_ = copy.filters_.push_back(filter);
+  }
+  return copy;
+}
```

Comment:
is this function really needed? what does the copy c'tor and the assignment operator do for `immutable::AppendOnlyList`?

Reply:
Use tests to verify the behaviour of  `immutable::AppendOnlyList`. Changing the filter inside `test_query` won't affect `query_`. Will delete this method and use copy c'tor instead.